### PR TITLE
.github/pr-license.yaml: add license scan for python

### LINKED ISSUE #5: add github action for license checker

### PR DESCRIPTION
**Pull request by hairongchen**
This PR is to add python license check in github action to fix #5
Golang and Rust license check will be added with corresponding codes check-in

Signed-off-by: hairongchen [hairong.chen@intel.com](mailto:hairong.chen@intel.com)

### CODE REVIEW

File: .github/workflows/pr-license.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  workflow_dispatch:
+
+jobs:
+  python-license-scan:
+    runs-on: self-hosted
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.10
```
Comment (leyao-daily) [inline .github/workflows/pr-license.yaml:14]: I think you can remove to use the default on the host.

File: .github/workflows/pr-license.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      - name: Check License
+        run: |
+          set -ex
+          python3 -m pip install liccheck
+          for f in $(find -type f -name "requirements.txt"); do
+            liccheck -s .github/.license_check.ini -r $f
+          done
```
Comment (leyao-daily) [inline .github/workflows/pr-license.yaml:21]: add newline at the end.

File: .github/.license_check.ini [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        mit license
+        mozilla public license 2.0 (mpl 2.0)
+        python software foundation
+        python software foundation license
+        zpl 2.1
+        GNU Lesser General Public License v2 or later (LGPLv2+)
+        MPL-2.0
+
+unauthorized_licenses:
+        gpl v3
```
Comment (leyao-daily) [inline .github/.license_check.ini:29]: add newline at the end. and for the license, i think it is better to sort them in order.

File: .github/.license_check.ini [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# Authorized and unauthorized licenses in LOWER CASE
+[Licenses]
+authorized_licenses:
+        3-clause bsd
```
Comment (leyao-daily) [inline .github/.license_check.ini:4]: It can be better to sort them alphabetically.

**Review by leyao-daily — APPROVED** [review]
LGTM. Thanks.

File: .github/workflows/pr-license.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+name: Check License
```
Comment (leyao-daily) [inline .github/workflows/pr-license.yaml:1]: Change to `License Check` to be same format with other action name in this project.

**Review by leyao-daily — APPROVED** [review]

File: .github/.license_check.ini [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        apache
+        apache 2.0
+        apache software
+        apache software license
+        apache-2.0
+        bsd
+        bsd license
+        GNU Lesser General Public License v2 or later (LGPLv2+)
```
Comment (hwang37) [inline .github/.license_check.ini:12]: Should we move gpl license to unauthorized license?
        GNU Lesser General Public License v2 or later (LGPLv2+)
        gpu lgpl
        lgpl with exceptions or zpl

Comment (kenplusplus) [reply .github/.license_check.ini:12]: LGPL is allowed

**Pull request merged**